ambry-router/src/test/java/com.github.ambry.router/GetOperationTest.java

```diff
@@ -0,0 +1,474 @@
+  private final int BLOB_SIZE = 100;
+  private final int BLOB_USER_METADATA_SIZE = 10;
+  private final Random random = new Random();
+  private final String blobIdStr;
+  private final BlobProperties blobProperties;
+  private final byte[] userMetadata;
+  private final byte[] putContent;
+  private final GetTestRequestRegistrationCallbackImpl requestRegistrationCallback =
+      new GetTestRequestRegistrationCallbackImpl();
+  private final FutureResult<BlobInfo> operationFuture = new FutureResult<>();
+  private final AtomicReference<BlobInfo> operationResult = new AtomicReference<>(null);
+  private final AtomicReference<Exception> operationException = new AtomicReference<>(null);
+  private final Callback<BlobInfo> operationCallback = new Callback<BlobInfo>() {
+    @Override
+    public void onCompletion(BlobInfo result, Exception exception) {
+      operationResult.set(result);
+      operationException.set(exception);
+    }
+  };
+
+  private class GetTestRequestRegistrationCallbackImpl implements RequestRegistrationCallback<GetOperation> {
+    private List<RequestInfo> requestListToFill;
+
+    @Override
+    public void registerRequestToSend(GetOperation getOperation, RequestInfo requestInfo) {
+      requestListToFill.add(requestInfo);
+      correlationIdToGetOperation.put(((RequestOrResponse) requestInfo.getRequest()).getCorrelationId(), getOperation);
+    }
+  }
+
+  ;
+
+  public GetOperationTest()
+      throws Exception {
+    VerifiableProperties vprops = getNonBlockingRouterProperties();
+    routerConfig = new RouterConfig(vprops);
+    mockClusterMap = new MockClusterMap();
+    mockServerLayout = new MockServerLayout(mockClusterMap);
+    replicasCount = mockClusterMap.getWritablePartitionIds().get(0).getReplicaIds().size();
+    responseHandler = new ResponseHandler(mockClusterMap);
+    time = new MockTime();
+    networkClientFactory = new MockNetworkClientFactory(vprops, mockSelectorState, MAX_PORTS_PLAIN_TEXT, MAX_PORTS_SSL,
+        CHECKOUT_TIMEOUT_MS, mockServerLayout, time);
+    router = new NonBlockingRouter(new RouterConfig(vprops), new NonBlockingRouterMetrics(new MetricRegistry()),
+        networkClientFactory, new LoggingNotificationSystem(), mockClusterMap, time);
+    blobProperties = new BlobProperties(BLOB_SIZE, "serviceId", "memberId", "contentType", false, Utils.Infinite_Time);
+    userMetadata = new byte[BLOB_USER_METADATA_SIZE];
+    random.nextBytes(userMetadata);
+    putContent = new byte[BLOB_SIZE];
+    random.nextBytes(putContent);
+    ReadableStreamChannel putChannel = new ByteBufferReadableStreamChannel(ByteBuffer.wrap(putContent));
+    blobIdStr = router.putBlob(blobProperties, userMetadata, putChannel).get();
+    router.close();
+    networkClient = networkClientFactory.getNetworkClient();
+  }
+
+  @Before
+  public void before() {
+    // almost all tests create and complete an operation outside of the router. Router's count of active operations
```

Comment:
this seems to be working under the assumption that each test *will surely* have an operation done outside of the `Router`. Otherwise it has to "fix" this variable before the test ends.

I'm not sure I agree with this - the check itself is reduced to something that is not reliable.